### PR TITLE
Allow box zoom in any direction for plots

### PR DESCRIPTION
**Pull request by Zoxc**
Previously only the top-left to bottom-right direction worked.

### CODE REVIEW

File: crates/egui/src/widgets/plot/mod.rs [hunk 1]
```diff
@@ -876,14 +876,18 @@ impl Plot {
...
+                        max: [
+                            box_start_pos.x.max(box_end_pos.x),
+                            box_start_pos.y.max(box_end_pos.y),
+                        ],
                     };
                     if new_bounds.is_valid() {
                         transform.set_bounds(new_bounds);
                         auto_bounds = false.into();
-                    } else {
-                        auto_bounds = true.into();
```
Comment (emilk) [inline crates/egui/src/widgets/plot/mod.rs:886]: why this change?

Comment (emilk) [reply crates/egui/src/widgets/plot/mod.rs:886]: If you don't know why it was there, maybe it is not a good idea to remove it? :)

Comment (Zoxc) [reply crates/egui/src/widgets/plot/mod.rs:886]: I don't want box zoom to ever trigger auto bounds, we already have double clicking to reset the view.
